### PR TITLE
Add WA SNAP

### PR DESCRIPTION
**Pull request by patwey**
## Context & Motivation

Implements WA Basic Food (SNAP) as a PolicyEngine-backed program for the Washington white label.

Linear: https://linear.app/myfriendben/issue/MFB-848/wa-snap-implementation

## Changes Made

- Added `WaSnap` calculator class (`programs/programs/wa/pe/spm.py`) — extends the federal `Snap` calculator with `WaStateCodeDependency` so PolicyEngine applies WA-specific SNAP rules (200% FPL gross income limit via WA's Broad-Based Categorical Eligibility)
- Registered `wa_snap` in `programs/programs/wa/pe/__init__.py` and the central `registry.py`
- Added unit tests for `WaSnap` (`programs/programs/wa/pe/tests/test_spm.py`)
- Added research files: spec, initial config, and test cases
- Added `wa` to the validations JSON schema (`test_case_schema.json`)
- Fixed test case values from monthly ($72) to annual ($864) to match system convention

## Testing

- Migrations to run: none
- Configuration updates needed: none
- Manual testing steps:
  1. Run `python manage.py validate --white-label wa` — all 6 validations pass (Passed: 6, Failed: 0, Skipped: 0)
  2. Run unit tests: `python manage.py test programs.programs.wa.pe.tests`

## Deployment

- Post-deployment scripts needed: Run `python manage.py import_program_config programs/management/commands/import_program_config_data/data/wa_snap_initial_config.json` if not already run on the target environment
- Admin updates needed: Ensure `wa_snap` program has `active = True` in the WA white label admin

## Notes for Reviewers

- `WaSnap` follows the same pattern as `TxSnap`, `CoSnap`, `NcSnap`, `IlSnap` — extends federal `Snap` with a state code dependency
- WA uses Broad-Based Categorical Eligibility (BBCE) which raises the gross income limit to 200% FPL (vs. standard 130% FPL) — PolicyEngine handles this when `state_code=WA` is provided
- Test case values are annual (monthly PE output × 12), consistent with how all other SNAP programs report values
- Known limitation: elderly/disabled alternative eligibility path (criterion 7 in spec) is not evaluated in the screener — a disclaimer is recommended for affected households

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added Washington SNAP (Basic Food) support: eligibility screening, benefit estimation, and registry integration.

* **Documentation**
  * Added comprehensive WA SNAP spec and a QA report summarizing validation scenarios and results.

* **Tests**
  * Added unit tests covering WA SNAP calculator behavior and screening mappings.

* **Chores**
  * Added WA SNAP initial config and validation fixtures; updated import schema to accept the WA white-label.

### CODE REVIEW

File: programs/programs/wa/snap/spec.md [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+**Steps**:
+- **Location**: Enter ZIP code `98103`, Select county `King`
+- **Household**: Number of people: `1`
+- **Person 1**: Birth month/year: `June 1991` (age 34), Relationship: Head of Household, Sex: Male, Not a student, Not pregnant, No disability, US citizen, Employment income: `$2,661` per month ($1 above 200% FPL for HH of 1 in 2026: 200% FPL = $2,660/mo)
+- **Expenses**: Rent/mortgage: `$900` per month, No other deductions or expenses
+- **Current Benefits**: Not currently receiving SNAP/Basic Food, Not receiving TANF, Not receiving SSI
+
+**Why this matters**: This tests the upper boundary of the gross income test under Washington's BBCE policy. A household exceeding this threshold by even $1 should be denied, confirming the screener correctly enforces the boundary rather than rounding or allowing a tolerance. Note: income updated from original spec to reflect 2026 FPL values (200% FPL for HH of 1 = $2,660/mo; use $2,661 to be $1 above).
+
+---
```
Comment (catonph) [inline programs/programs/wa/snap/spec.md:251]: **FPL year inconsistency — needs clarification**

Scenarios 3 and 4 correctly use SNAP FY2026 thresholds based on the *2025 FPL* (effective Oct 2025–Sep 2026): HH=2 → 200% FPL = $3,525/mo. But this scenario uses $2,660/mo for HH=1, which matches the *2026 calendar-year FPL* ($15,960 × 2 / 12 = $2,660) — not the 2025 FPL that governs SNAP FY2026 ($15,060 × 2 / 12 = **$2,510/mo**).

If PolicyEngine is using the Oct 2025–Sep 2026 thresholds (as it should for the current benefit year), then $2,661 is not $1 above the actual limit — it's well below it. The boundary test won't catch what it's intended to catch.

Can you confirm which FPL year PolicyEngine is using for WA SNAP right now, and align all scenarios to the same base year? The changelog says "updated to 2026 FPL values" but scenarios 3/4 and scenario 5 are using different years.

File: programs/programs/wa/snap/spec.md [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+**Steps**:
+- **Location**: Enter ZIP code `98101`, Select county `King`
+- **Household**: Number of people: `1`
+- **Person 1**: Birth month/year: `January 1951` (age 75), Relationship: Head of Household, Not a student, Not pregnant, Not disabled (using age 60+ exemption, not disability)
+- **Income**: Social Security Retirement income: `$1,200` per month, No other income sources
+- **Current Benefits**: Not currently receiving SNAP/Basic Food, Not receiving TANF, Not receiving SSI
+
+**Why this matters**: Confirms that a person well above the minimum age threshold is correctly recognized as eligible. Also validates the critical elderly exemption (age 60+) pathway. At $1,200/mo income for HH of 1, this household is well below 200% FPL ($2,660/mo), so it qualifies directly via the gross income test without needing the alternative net-income-only path.
+
+---
```
Comment (catonph) [inline programs/programs/wa/snap/spec.md:303]: **Scenario 8 doesn't exercise the elderly alternative path**

At $1,200/mo income for HH=1, this person passes the 200% FPL gross income test directly ($2,660 threshold), so criterion 7's alternative path never triggers. This scenario validates a low-income elderly person is eligible — but so does scenario 1. The distinctive thing about elderly/disabled households under criterion 7 is the *Option B path*: gross income > 200% FPL but net income ≤ 100% FPL still qualifies.

Consider updating scenario 8 to set gross income above 200% FPL but net income ≤ 100% FPL (e.g. $2,800/mo gross, $800/mo rent → net income well under 100% FPL threshold of $1,255/mo for HH=1). That would actually test the edge case this scenario is named for.

File: programs/programs/wa/snap/spec.md [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+
+**Steps**:
+- **Location**: Enter ZIP code `98101`, Select county `King`
+- **Household**: Number of people: `1`
+- **Person 1**: Birth month/year: `June 1991` (age 34), Relationship: Head of Household, No disability, Not pregnant, Not a student enrolled in higher education, U.S. citizen
+- **Income**: Employment income: `$1,200` per month, No other income sources
+- **Current Benefits**: Not currently receiving SNAP/Basic Food, Not receiving TANF or SSI
+
+**Why this matters**: Confirms that a valid Washington State ZIP code (98101 — downtown Seattle, King County) is properly recognized as within the Basic Food service area. Geographic validation is essential because SNAP is state-administered and applicants must reside in the state where they apply (7 CFR 273.3; WAC 388-468-0005).
+
```
Comment (catonph) [inline programs/programs/wa/snap/spec.md:320]: **Scenario 9 is redundant — every other eligible scenario already validates WA residency**

All eligible scenarios use a WA ZIP code (98101/98103), so geographic validation is implicitly covered. This slot would be more valuable as a test for one an unautomated criteria:

- **Net income test failure** — a household that passes the 200% FPL gross test but fails the 100% FPL net test (currently no scenario isolates this)

File: programs/programs/wa/snap/spec.md [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+**Steps**:
+- **Location**: Enter ZIP code `98103`, Select county `King`
+- **Household**: Number of people: `2`
+- **Person 1**: Birth month/year: `June 1986` (age 39), Relationship: Head of Household, No disability, Not pregnant, Not a student, Employment income: `$1,800` per month
+- **Person 2**: Birth month/year: `September 2018` (age 7), Relationship: Child, No disability
+- **Current Benefits**: Indicate that the household is currently receiving SNAP/Basic Food (`has_snap` = Yes)
+
+**Why this matters**: This scenario differs from Scenario 10 by using a parent-child household rather than a single adult, ensuring the duplicate benefit exclusion applies regardless of household composition.
+
+---
```
Comment (catonph) [inline programs/programs/wa/snap/spec.md]: **Scenarios 10 and 11 are redundant — the `has_snap` exclusion doesn't vary by household composition**

The duplicate-benefit exclusion in `_build_benefit_map` maps `wa_snap → self.has_snap` — it doesn't inspect household composition. Both scenarios set `has_snap=True` and expect ineligible; the only difference is 1 adult vs. parent+child. One test case covers this logic; the second doesn't add regression protection.

Suggest replacing scenario 11 with a net-income-test-failure case: household that passes the 200% FPL gross income test but has net income above 100% FPL (e.g. HH=1, gross income $2,000/mo with no deductions → net income $2,000 >> 100% FPL threshold of ~$1,255/mo → ineligible). Criterion 2 has no dedicated test.

**Review by catonph — COMMENTED** [review]
Test scenario and FPL values review — see inline comments below.

File: validations/management/commands/import_validations/data/wa_snap.json [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    "expected_results": {
+      "program_name": "wa_snap",
+      "eligible": true,
+      "value": 864
+    }
+  },
+  {
+    "notes": "WA Basic Food (SNAP) - Single Adult, Gross Income $1 Above 200% FPL ($2,661/mo) - Should NOT Be Eligible. Patmanson correction: 2026 200% FPL for 1 person is $2,660/mo; original scenario incorrectly used $2,431.",
+    "household": {
+      "white_label": "wa",
```
Comment (catonph) [inline validations/management/commands/import_validations/data/wa_snap.json:42]: **Same FPL year question as in the spec — $2,660 vs $2,510**

The note here says "2026 200% FPL for 1 person is $2,660/mo" but spec scenarios 3/4 use $3,525/mo as 200% FPL for HH=2 based on the *2025 FPL* (SNAP FY2026). Those two are from different FPL base years:

- 2025 FPL (governs SNAP Oct 2025–Sep 2026): 100% FPL for 1 person = $15,060/yr → 200% = $2,510/mo
- 2026 calendar-year FPL: 100% FPL for 1 person = $15,960/yr → 200% = $2,660/mo

If PolicyEngine is using SNAP FY2026 thresholds (2025 FPL base), then $2,661 is not above the limit and this test case will return eligible instead of ineligible. Please confirm which year PolicyEngine is keyed to and adjust `amount` accordingly.

**Review by catonph — CHANGES_REQUESTED** [review]
A few test scenario suggestions / FPL consistency needed

**Comment by patwey** [comment]
@coderabbitai review

File: programs/management/commands/import_program_config_data/data/wa_snap_initial_config.json [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    "year": "2026",
+    "legal_status_required": [
+      "citizen",
+      "gc_5plus",
+      "gc_5less",
+      "refugee",
+      "otherWithWorkPermission"
+    ],
+    "name": "Basic Food - Supplemental Nutrition Assistance Program (SNAP)",
+    "description": "Basic Food (also called SNAP or food stamps) helps you buy groceries each month. You get a card that works like a debit card at most grocery stores. The amount you get depends on your household size and income.\n\nMost people with low income can apply. Washington has higher income limits than many states. For most people, you do not need to meet an asset or savings test. If you get TANF (cash help) or SSI (disability benefits), you may qualify right away. College students may need to work at least 20 hours a week to qualify.",
```
[the remark this answers is not in the document]
Comment (patwey) [reply programs/management/commands/import_program_config_data/data/wa_snap_initial_config.json:19]: Thanks for the suggestion! Since ineligible users never see the program description, a disclaimer there wouldn't reach the people it's meant to help. No change here.

File: programs/management/commands/import_program_config_data/data/wa_snap_initial_config.json [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    "description_short": "Monthly help buying groceries for low-income households",
+    "learn_more_link": "https://www.dshs.wa.gov/esa/community-services-offices/basic-food",
+    "apply_button_link": "https://www.washingtonconnection.org/home/",
```
[the remark this answers is not in the document]
Comment (patwey) [reply programs/management/commands/import_program_config_data/data/wa_snap_initial_config.json:22]: WashingtonConnection is the official WA state portal for submitting a SNAP application, so it's the right destination for `apply_button_link`. The HelpMeGrow link is already included as the navigator's `assistance_link` — these serve different purposes. No change here.

File: programs/programs/wa/snap/spec.md [hunk 8]
```diff
@@ -0,0 +1,420 @@
...
+**Steps**:
+- **Location**: Enter ZIP code `98101`, Select county `King`
+- **Household**: Number of people: `1`
+- **Person 1**: Birth month/year: `June 1991` (age 34), Relationship: Head of Household, Not a student enrolled in higher education, Not pregnant, No disability, U.S. citizen
+- **Income**: Employment income: `$2,200` per month, No other income sources
+- **Expenses**: No significant shelter or dependent care expenses
+- **Current Benefits**: Not currently receiving SNAP/Basic Food, Not receiving TANF or SSI
+
+**Why this matters**: Tests the net income test (criterion 2) as a standalone ineligible gate. Gross income ($2,200/mo) is well under the 200% FPL limit ($2,660/mo), so the household passes the gross test. After applying the 20% earned income deduction ($440) and the FY2025 standard deduction for HH of 1 ($219), net income is approximately $1,541/mo — above the 100% FPL net limit of $1,330/mo — making the household ineligible. Without significant shelter or dependent care costs, no additional deductions reduce net income below the threshold. Every other eligible scenario in this spec already uses a WA ZIP code, so geographic validation is implicitly covered.
+
```
[the remark this answers is not in the document]
Comment (patwey) [reply programs/programs/wa/snap/spec.md]: Good catch that there are wrong numbers, but the correct values differ from the suggestion. PolicyEngine's `snap_fpg` variable uses SNAP fiscal year logic — for any month Jan–Sep it looks up the HHS FPL as of October 1 of the prior year. Since this program calculates for period `2026-01`, PE resolves to the 2025 HHS guidelines ($15,650/yr), giving 100% FPL ≈ $1,304/mo and 200% FPL ≈ $2,608/mo. Updating all three incorrect references in Scenarios 8 and 9 accordingly. The Scenario 9 "FY2025 standard deduction" concern is already moot — that scenario was rewritten and no longer references a standard deduction.

File: programs/programs/wa/snap/spec.md [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+**Steps**:
+- **Location**: Enter ZIP code `98103`, Select county `King`
+- **Household**: Number of people: `3`
+- **Person 1**: Birth month/year: `June 1960` (age 65), Relationship: Head of Household, Not a student, Social Security Retirement income: `$1,200` per month, Not currently receiving SNAP/Basic Food
+- **Person 2**: Birth month/year: `September 1996` (age 29), Relationship: Child/Dependent adult, Not a student, No disability, Employment income: `$2,400` per month
+- **Person 3**: Birth month/year: `January 2004` (age 22), Relationship: Grandchild, Student: Yes (enrolled at least half-time in higher education), Works 20+ hours per week (qualifies for student exemption), Employment income: `$900` per month, No disability
+- **Current Benefits**: Not currently receiving SNAP/Basic Food, Not receiving TANF or SSI
+
+**Why this matters**: Validates that the screener correctly applies the elderly/disabled gross income exemption while still enforcing the net income test, and properly handles student eligibility rules within a multi-member household.
+
```
[the remark this answers is not in the document]
Comment (patwey) [reply programs/programs/wa/snap/spec.md:355]: Confirmed — good catch. Gross income ($1,200 + $2,400 + $900 = $4,500/mo) exceeded the BBCE 200% FPL threshold for HH=3 ($4,442/mo at 2025 HHS guidelines). Additionally, PolicyEngine's WA SNAP implementation (`is_snap_eligible.py`) treats gross > 200% FPL as a hard disqualification for all WA households — there is no Option B path in PolicyEngine for WA; `net_applies/hheod.yaml: WA: false` only waives the net test when the gross test *passes*, not when it fails. Fixed by reducing Person 2's income from $2,400 to $1,500/mo — total gross is now $3,600/mo, clearing the BBCE threshold and making the household categorically eligible with the net income test waived. The scenario still exercises criteria 1, 2, 4, 6, and 7 (Option A path for the elderly member). Local QA confirmed eligible.

File: programs/programs/wa/snap/spec.md [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+Updated `eligible` values:
+- Scenarios 1, 2, 3, 4, 6, 7, 8, 9, 11, 12: `true`
+- Scenarios 5, 10: `false`
+
+Updated income amounts for 2026 FPL:
+- Scenario 3 (HH=2, $1 below 200% FPL): `$3,606/mo` (was $2,429)
+- Scenario 4 (HH=2, exactly 200% FPL): `$3,607/mo` total combined wages (was $2,510); net income at exactly 100% FPL = `$1,803/mo`
+- Scenario 5 (HH=1, $1 above 200% FPL): `$2,661/mo` (was $2,431)
+
```
[the remark this answers is not in the document]
Comment (patwey) [reply programs/programs/wa/snap/spec.md]: Confirmed — this block was written during the 2026-04-07 FPL update and not removed during the 2026-04-24 revert to 2025 HHS guidelines. The actual scenario values ($3,524/$3,525 for scenarios 3/4) are already documented inline in each scenario and in the changelog, which explains the SNAP fiscal-year reasoning (`snap_fpg.py`: Jan–Sep of year Y uses prior Oct 1 parameters). Removed the stale summary to eliminate the contradiction.

File: programs/programs/wa/snap/spec.md [hunk 11]
```diff
@@ -0,0 +1,416 @@
...
+7. **Elderly or disabled household member — alternative eligibility path**
+   - Screener fields: `age`, `disabled`, `calc_gross_income`, `calc_net_income`, `household_assets`
+   - Note: Households with a member aged 60+ or with a disability have **two paths** to eligibility:
+     - **Option A (pass gross income test)**: Gross income ≤ 200% FPL → eligible; net income test is not required.
+     - **Option B (fail gross income test)**: Gross income > 200% FPL → can still be eligible if net income ≤ 100% FPL. A $4,500 asset limit also applies under this path.
+   - **Implementation**: Evaluate Option B in the screener. Check: (1) household has a member aged 60+ or with a disability, (2) gross income > 200% FPL, (3) net income ≤ 100% FPL, (4) assets ≤ $4,500. If all four conditions are met → eligible. Note: the screener approximates net income as gross minus housing/care expenses; the actual SNAP deduction formula (standard deduction, excess shelter, earned income deduction) is more complex — see criterion 21.
+   - Source: 7 CFR 273.9(b)(1); 7 U.S.C. § 2014(c)(1); WAC 388-470-0005.
```
[the remark this answers is not in the document]
Comment (patwey) [reply programs/programs/wa/snap/spec.md:45]: After tracing through the PolicyEngine code, this concern is based on a conflation of two separate eligibility paths.

**The BBCE (categorical eligibility) path** — `meets_tanf_non_cash_gross_income_test` — does treat gross > 200% FPL as a hard cutoff. An elderly household with gross > 200% FPL cannot qualify via categorical eligibility.

**The normal SNAP eligibility path** — `meets_snap_gross_income_test` (`policyengine_us/variables/gov/usda/snap/eligibility/meets_snap_gross_income_test.py`) — is different:

```python
has_elderly_disabled = spm_unit("has_usda_elderly_disabled", period)
return has_elderly_disabled | (income <= limit)
```

The gross income test is **waived** for HHEOD households at the federal level. There is no hard disqualification.

**Tracing Scenario 8** (age 75, gross $2,700/mo > $2,608 FPL, net $1,200/mo < $1,304 FPL, assets $0):
- BBCE gross test: $2,700 > 200% FPL → fails categorical eligibility
- Normal gross test: `has_elderly_disabled = True` → **waived** ✅
- Normal net test: $1,200/$1,304 ≈ 0.92 ≤ 1.0 → **passes** ✅
- Asset test: $0 ≤ $4,500 (elderly limit) → **passes** ✅
- Result: **Eligible via `normal_eligibility`** — exactly what Option B describes

**What `net_applies.hheod[WA] = false` means**: this parameter only appears in `meets_tanf_non_cash_net_income_test` — the BBCE path. Since Scenario 8 fails the BBCE gross test first, this WA parameter is never reached and has no bearing on Option B.

There is no WA-specific override of `meets_snap_gross_income_test` or `is_snap_eligible`. The spec, Scenario 8 expected result, criterion 7 wording, coverage count of 11, and changelog are all correct. No changes needed.

**Review by catonph — APPROVED** [review]
LGTM!
